### DISCUSSION TITLE
Support for request/response types

### DISCUSSION THREAD

**Discussion by bwgjoseph** [feathersjs/feathers#2218] [Ideas]
Hi!

Would like to know whether this would be considered in the upcoming v5 release or further roadmap? 

I'm quite curious/surprised that this hasn't been brought up (at all, I think). To me, there are 

1. ClientRequestInterface
2. ServerRequestInterface
3. ResponseInterface

See https://github.com/feathersjs/feathers/pull/2125 for more detailed info. I realize my PR might still be lacking something, but am open for further discussion if this is something that can be considered

**Comment by daffl** [comment]
Yes, this will be a part of https://github.com/feathersjs/feathers/tree/ts-refactor-dove. For Hook types, it should also be possible to infer the generics from the service interface and/or application types.

**Reply by bwgjoseph** [reply]
That's really great to hear! 

I briefly looked at the branch you mentioned, and didn't see the r/r types defined yet so I assume that isn't in yet. Will be looking forward to it, and try it out once you have something ready to be tested.

**Reply by daffl** [reply]
These changes have now been published in v5.0.0-beta.0 - the migration guide shows [how to test the prerelease](https://dove.docs.feathersjs.com/guides/migrating.html#testing-the-prerelease). The docs definitely need a TypeScript usage page added.
